services/motion/builtin/builtin_test.go

```diff
@@ -113,14 +115,44 @@ func createMoveOnGlobeEnvironment(ctx context.Context, t *testing.T, gpsPoint *g
 		nil,
 	)
 
+	// create a fake kinematic base
+	localizer := motion.NewMovementSensorLocalizer(staticMovementSensor, origin, spatialmath.NewZeroPose())
+	straightlineDistance := spatialmath.GeoPointToPose(destination, origin).Point().Norm()
+	limits := []referenceframe.Limit{
+		{Min: -straightlineDistance * 3, Max: straightlineDistance * 3},
+		{Min: -straightlineDistance * 3, Max: straightlineDistance * 3},
+		{Min: -2 * math.Pi, Max: 2 * math.Pi},
+	}
+	kinematicsOptions := kinematicbase.NewKinematicBaseOptions()
+	kinematicsOptions.PlanDeviationThresholdMM = 1 // can afford to do this for tests
+	kb, err := kinematicbase.WrapWithFakeKinematics(ctx, fakeBase.(*fake.Base), localizer, limits, kinematicsOptions)
+	test.That(t, err, test.ShouldBeNil)
+
+	// create injected MovementSensor
+	dynamicMovementSensor := inject.NewMovementSensor("test-gps")
+	dynamicMovementSensor.PositionFunc = func(ctx context.Context, extra map[string]interface{}) (*geo.Point, float64, error) {
+		input, err := kb.CurrentInputs(ctx)
+		test.That(t, err, test.ShouldBeNil)
+		heading := rdkutils.RadToDeg(math.Atan2(input[0].Value, input[1].Value))
+		distance := math.Sqrt(input[1].Value*input[1].Value + input[0].Value*input[0].Value)
+		pt := origin.PointAtDistanceAndBearing(distance*1e-6, heading)
```

Comment:
[q] do you multiply distance by 1e-6 to convert mm to km

Reply:
yes